### DISCUSSION TITLE
Content Management System Discussion

### DISCUSSION THREAD

**Discussion by WillieCubed** [acmutd/nebula-guide#2] [Ideas]
The UTD Survival Guide needs a content management system that meets the following criteria:
- Usability: able to create, read, update, and delete content without needing to modify source code
- Extensibility: able to easily add new data object types and functionality to the guide
- Scalability: able to support dozens of different data types (including multimedia like photos and video) and hundreds of different objects without sacrificing performance

@VishvakBandi, give your assessment for CMSes below.

**Comment by adityarathod** [comment]
A suggestion:

[Netlify CMS](https://www.netlifycms.org/) is a Git-based CMS that works with any static site generator. It satisfies all your requirements:

- Usability: WYSIWYG editor with ability to manage posts and metadata
- Extensibility: Supports various widget types to record varied data types per data object
- Scalability: Supports everything that Git supports, but may need a media host for much larger files

The only caveat I see is that Netlify CMS needs to link to a centralized identity provider (like Netlify Identity, whose free tier supports 1k users), and that may not be something you want.

**Comment by VishvakBandi** [comment]
I did some research on this earlier this week, and I mostly agree with @adityarathod. NetlifyCMS is definitely the best.

Here are the CMSes I looked at.
- Netlify CMS - I think Netlify meets all of our needs. It's not overly complicated, nor does it restrict us broadly on how we can create or host content. The main admin console is extremely user-friendly and easy to maintain - the biggest challenge would be organizing the content within the CMS. It also supports a lot of different content types as Aditya mentioned. Everything content-wise is hosted on the same git repository, so if media does become large, we will need to host it elsewhere. 
  - Having everything on the git repository is also great because contributors can just make a pull request to add content (although they can use the console as well). 
  - Netlify rebuilds at every commit to the master branch, which will burn through build minutes pretty quickly. To get around this, we can only modify the content at set intervals by editing it on another branch and merging when needed.
- [ghost](https://ghost.org/) - Ghost is a great CMS and extremely powerful, but definitely too powerful and customizable for our problem. Configuring this would take significant amounts of time. The CMS is also API-based, so hosting and deployment will be a significantly higher overhead compared to Netlify.
- [Tina](https://tina.io/) - This is a git-based API, but has its shortcomings with hosting (Netlify's integrations are very good). 
- I took a look at a few more, but they were not as good as Netlify. Most were plugin-based for existing platforms or relied on a lot more setup and maintenance from developers.

Overall, I do think that Netlify CMS meets all of our needs without constricting the project in a certain way. If we ever change our minds, it seems pretty easy to move to another git-based CMS, but some more work will be required to move to one like ghost, though I do not anticipate a need to do that.
